### DISCUSSION TITLE
Email templates, where?

### DISCUSSION THREAD

**Discussion by anvaya** [jasontaylordev/CleanArchitecture#1073] [Q&A]
In a scenario where the system is supposed to acknowledge a customer request by sending the customer an email and at the same time send the staff an alert about the same, where would you put your email templates and how would you process them?

I would assume the Application project should do that with its event handlers, however the email templates would need variable processing (replacement) through either custom or razor templating.  Another issue is of sharing the design with the web UI project (logos, theme) etc. 

Has anybody addressed this issue, how have you gone about it?

**Comment by 1988gadocansey** [comment]
Hi, the email processing will be placed in the infrastructure layer.
[image: Screen Shot 2023-12-18 at 8.11.08 AM.png]
My little contribution.

**Reply by anvaya** [reply]
Thank you, yes more or less, and there is a discussion topic already on that matter with a detailed answer. My concern was more about templates and where one would place them. 

https://github.com/jasontaylordev/CleanArchitecture/discussions/685

The answer given indicated to use the WebUI layer for the templates, however in our case the plan is to use some background service (or queue processor) that processes emails so that we can keep the API breezy and also make sure retries happen in case of failures, so it wont technically be the WebUI project that would directly trigger email sending. 

This gets further complicated if you use an Web API project with Rich client like Blazor, React or Angular, and host the client separately (static hosting) for example, and hence the UI project would actually be located somewhere else.
